src/TransferList/TransferList.js

```diff
@@ -0,0 +1,228 @@
+              return (
+                <ListItem
+                  key={value}
+                  role="listitem1"
+                  button
+                  onClick={() => handleToggle(value)}
+                  disablePadding
+                >
+                  <ListItemIcon>
+                    <Checkbox
+                      checked={selectedItems.indexOf(value) !== -1}
+                      disableRipple
+                    />
+                  </ListItemIcon>
+                  <ListItemText primary={value} />
+                </ListItem>
+              );
+            })}
+            <ListItem />
+          </List>
+        </Box>
+      </Box>
+      <Box
+        sx={{
+          display: "flex",
+          flexDirection: "column",
+          height: "100%",
+          width: "100%"
+        }}
+      >
+        <Box
+          sx={{
+            borderBottom: theme => `1px solid ${theme.palette.divider}`,
+            disply: "flex",
+            flexDirection: "column",
+            height: "40px",
+            position: "relative",
+            width: "100%"
+          }}
+        >
+          {selectedItems.length === 0 ? (
+            <Typography data-testid="none-selected" sx={{ pl: 1, pt: 1 }}>
+              None Selected
+            </Typography>
+          ) : (
+            <>
+              <Typography sx={{ pl: 1, pt: 1 }}>
+                {`${selectedItems.length} selected`}
+              </Typography>
+              <Button
+                data-testid="clear-all"
+                variant="text"
+                sx={{ position: "absolute", right: 2, top: 2 }}
+                onClick={() => onChange([])}
+              >
+                Clear All
+              </Button>
+            </>
+          )}
+        </Box>
+        <Box
+          sx={{
+            height: "100%",
+            width: "100%"
+          }}
+        >
+          <List dense component="div" role="sortedlist">
+            {selectedItems.map(value => {
+              return (
+                <ListItem
+                  key={value}
+                  role="listitem2"
+                  onClick={() => handleToggle(value)}
+                  secondaryAction={
+                    <IconButton edge="end">
+                      <CloseIcon data-testid="close" />
+                    </IconButton>
+                  }
+                >
+                  <ListItemText primary={value} />
+                </ListItem>
+              );
+            })}
+            <ListItem />
+          </List>
+        </Box>
+      </Box>
+    </Box>
+  );
+}
+TransferList.propTypes = {
+  /**
+   * Array of Items.
+   */
+  items: PropTypes.arrayOf(
+    PropTypes.oneOfType([PropTypes.string, PropTypes.number])
+  ),
+  /**
+   * Callback fired when the new item is selected.
+   *
+   * **Signature**
+   * ```
+   * function(newItems) => void
+   * ```
+   *
+   * _newItems_: The new item that has been selected
```

Comment:
Callback contains all items selected, not just the new item that was clicked.

```suggestion
   * _newItems_: The new items that have been selected
```